crates/ln-dlc-node/src/ln/coordinator_event_handler.rs

```diff
@@ -0,0 +1,216 @@
+use super::event_handler::handlers;
+use super::event_handler::EventSender;
+use super::event_handler::PendingInterceptedHtlcs;
+use crate::node::Node;
+use crate::node::Storage;
+use crate::EventHandlerTrait;
+use anyhow::Result;
+use async_trait::async_trait;
+use bitcoin::hashes::hex::ToHex;
+use lightning::util::events::Event;
+use parking_lot::Mutex;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// Event handler for the coordinator node.
+// TODO: Move it out of this crate
+pub struct CoordinatorEventHandler<S> {
+    pub(crate) node: Arc<Node<S>>,
+    pub(crate) pending_intercepted_htlcs: PendingInterceptedHtlcs,
+    pub(crate) event_sender: Option<EventSender>,
+}
+
+impl<S> CoordinatorEventHandler<S>
+where
+    S: Storage + Sync + Send + 'static,
+{
+    pub fn new(node: Arc<Node<S>>, event_sender: Option<EventSender>) -> Self {
+        Self {
+            node,
+            event_sender,
+            pending_intercepted_htlcs: Arc::new(Mutex::new(HashMap::new())),
+        }
+    }
+}
+
+#[async_trait]
+impl<S> EventHandlerTrait for CoordinatorEventHandler<S>
+where
+    S: Storage + Send + Sync + 'static,
+{
+    fn event_sender(&self) -> &Option<EventSender> {
+        &self.event_sender
+    }
+
+    async fn match_event(&self, event: Event) -> Result<()> {
+        match event {
+            Event::FundingGenerationReady {
+                temporary_channel_id,
+                counterparty_node_id,
+                channel_value_satoshis,
+                output_script,
+                user_channel_id,
+            } => {
+                handlers::handle_funding_generation_ready(
+                    &self.node,
+                    user_channel_id,
+                    counterparty_node_id,
+                    output_script,
+                    channel_value_satoshis,
+                    temporary_channel_id,
+                )
+                .await?;
+            }
+            Event::PaymentClaimed {
+                payment_hash,
+                purpose,
+                amount_msat,
+                receiver_node_id: _,
+            } => {
+                handlers::handle_payment_claimed(&self.node, amount_msat, payment_hash, purpose);
+            }
+            Event::PaymentSent {
+                payment_preimage,
+                payment_hash,
+                fee_paid_msat,
+                ..
+            } => {
+                handlers::handle_payment_sent(
+                    &self.node,
+                    payment_hash,
+                    payment_preimage,
+                    fee_paid_msat,
+                )?;
+            }
+            Event::OpenChannelRequest {
+                temporary_channel_id,
+                counterparty_node_id,
+                funding_satoshis,
+                push_msat,
+                ..
+            } => {
+                handlers::handle_open_channel_request_trusted_0_conf(
```

Comment:
The coordinator should not accept 0-conf channels. Only the app should accept 0-conf channels from the coordinator. 

I guess we had this logic before?

Reply:
we didn't have any more logic, we only had a comment:
`TODO: only accept 0-conf from the coordinator.`

Reply:
#1075 fixes, but I can include it into this PR. it is the first behavioural change I've done, all the other stuff was "just" a refactor, ie. I tried to have zero behavioural changes.